src/mokka/src/main/java/pl/hycom/mokka/security/UserManager.java

```diff
@@ -41,34 +42,39 @@
 	@Autowired
 	private PasswordEncoder passwordEncoder;
 
-    public User createAdminUser(){
-	    log.info("Creating admin user.");
-        User user = new User();
-        user.setUserName("admin");
-        user.setPasswordHash(passwordEncoder.encode(getDefaultPassword()));
-        user.setFirstName("Admin");
-        user.setLastName("Admin");
-        user.setDisabled(Boolean.FALSE);
-        user.setRoles(ImmutableSet.of(Role.ADMIN));
+    public User createAdminUser() {
+        User userModel = createUserModel("admin", "Admin", "Admin", Role.ADMIN);
+        return userRepository.save(userModel);
+    }
+
+    public UserDetails createUser(String userName, DirContextOperations dirContextOperations) {
```

Comment:
I would be in favor of not including ldap related dependencies into `UserManager`.
I believe you should reuse `saveOrUpdateUser` (although I am not sure about the part related to password) or provide new one with `User` as an input parameter. 
DirContextOperations -> User conversion should be done on mapper side.

Reply:
I commited fixed version, please take a look.